tests/quick.rs

```diff
@@ -258,9 +258,7 @@ where
         let mut it = get_it();
 
         for _ in 0..(counts.len() - 1) {
-            if it.next().is_none() {
-                panic!("Iterator shouldn't be finished, may not be deterministic");
-            }
+            assert!(it.next().is_some(), "Iterator shouldn't be finished, may not be deterministic");
```

Comment:
Thinking about this a second time raises the question: Is it problematic that `it.next` appears within the `assert`? (In my ideal world, an `assert` should be side-effect free.)

Reply:
I can see your concern. In my world, side effects in `assert` is ok, in `debug_assert` it's bad (and there's a lint for that (written by me): https://rust-lang.github.io/rust-clippy/master/index.html#debug_assert_with_mut_call)

But I would understand if you would like to have that different. Please tell me.

Reply:
Would be fine for me. @jswrenn Do you have objections?